src/router.test.ts

```diff
@@ -248,6 +255,76 @@ describe('type inference', () => {
         }),
     );
   });
+
+  test('type inference when using Zod transforms', async () => {
+    const clients = setup((router) =>
+      router
+        .declare({
+          name: 'getItems',
+          route: 'GET /items',
+          request: z.object({
+            // API should enforce a string, but the code should receive a number.
+            message: z.string().transform((val) => Number(val)),
```

Comment:
Technically `NaN` would be returned by `Number(val)` in that case, so there would be no error I suppose.

Reply:
Good question. I'll get an example of what the error would look like if a `transform` throws.

Reply:
Thank you Swain. I just wanted to see if the UX would be uniform + correct here. Knowing you I bet it will be.

Reply:
@epeters3 Your question was important -- I uncovered something slightly unexpected (though, perhaps still non-blocking). It seems that _thrown_ errors in `.transform(...)` bubble up without being caught.

So, even though we use [`safeParse(...)`](https://zod.dev/?id=safeparse) to perform our parsing, that call will actually _throw_ if an underlying transform throws.

So, basically: it seems that Zod has taken the opinion of "transform functions must be safe".

--------
Separately, FWIW, it seems that using raw [`.transform(...)`](https://zod.dev/?id=transform) calls might not be best practice -- Zod documentation seems to instaed encourage the use of [`.pipe(...)`](https://zod.dev/?id=pipe) and [`.coerce`](https://zod.dev/?id=coercion-for-primitives):

```typescript
// instead of this:
z.string().transform(str => Number(str));

// do this:
z.string().pipe(z.coerce.number());
```

Doesn't _really_ do anything to solve our "thrown errors aren't caught" problem, but maybe indicates that using raw `.transform(...)` calls might be an uncommon case. Instead, we should be using primitives like `.coerce` and `.pipe` (which do provide more friendly errors).